modules/apps/headless-apio/blog/blog-apio-test/src/testIntegration/java/com/liferay/blog/apio/internal/architect/resource/test/BlogPostingNestedCollectionResourceTest.java

```diff
@@ -0,0 +1,195 @@
+import com.liferay.portal.apio.test.util.PaginationRequest;
+import com.liferay.portal.kernel.dao.orm.QueryDefinition;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionCheckerFactoryUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.test.context.ContextUserReplace;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Víctor Galán
+ */
+@RunWith(Arquillian.class)
+public class BlogPostingNestedCollectionResourceTest
+	extends BaseBlogPostingNestedCollectionResourceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testGetPageItems() throws Exception {
+		ServiceContext serviceContext = _getServiceContext(true);
+
+		BlogsEntry blogsEntry = _getBlogEntry(new Date(), serviceContext);
+
+		PageItems<BlogsEntry> pageItems = getPageItems(
+			PaginationRequest.of(10, 1), _group.getGroupId());
+
+		Assert.assertEquals(1, pageItems.getTotalCount());
+
+		Collection<BlogsEntry> blogEntries = pageItems.getItems();
+
+		Assert.assertTrue(
+			"Journal articles: " + blogEntries,
+			blogEntries.contains(blogsEntry));
+	}
+
+	@Test
+	public void testGetPageItemsWith1Pending() throws Exception {
+		ServiceContext serviceContext = _getServiceContext(true);
+
+		BlogsEntry blogsEntry = _getBlogEntry(new Date(), serviceContext);
+
+		_blogsEntryLocalService.updateStatus(
+			serviceContext.getUserId(), blogsEntry.getEntryId(),
+			WorkflowConstants.STATUS_PENDING, serviceContext,
+			Collections.emptyMap());
+
+		int blogEntriesCount = _blogsEntryLocalService.getGroupEntriesCount(
+			_group.getGroupId(), _anyStatusQueryDefinition);
+
+		Assert.assertEquals(1, blogEntriesCount);
```

Comment:
This method is tested in BlogsEntryLocalServiceTest, we don't need this extra assert.